pilot/pkg/model/validation.go

```diff
@@ -1338,7 +1346,10 @@ func ValidateVirtualService(name, namespace string, msg proto.Message) (errs err
 	allHostsValid := true
 	for _, host := range virtualService.Hosts {
 		if err := ValidateWildcardDomain(host); err != nil {
-			errs = appendErrors(errs, err)
+			ipAddr := net.ParseIP(host)
+			if ipAddr == nil { // Could be an IP address
+				errs = appendErrors(errs, err)
```

Comment:
what if its not nil? you are still marking allHostsValid=false..

Reply:
good catch ... fixed